src/main/java/com/github/checkstyle/regression/module/ModuleUtils.java

```diff
@@ -0,0 +1,202 @@
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle utility class.
+     */
+    public static boolean isCheckstyleUtility(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change);
+            returnValue = !NAME_TO_MODULE_EXTRACT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a Java main source file.
+     * Changes of Java main source files are which have path matching
+     * {@code JAVA_SOURCE_PARTTEN} and in "main" directory.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a Java main source file.
+     */
+    private static boolean isJavaMainSource(GitChange change) {
+        final boolean returnValue;
+        final Matcher matcher = JAVA_SOURCE_PARTTEN.matcher(change.getPath());
+        if (matcher.find()) {
+            returnValue = "main".equals(matcher.group(1));
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a test of checkstyle module.
+     * Changes of checkstyle module tests are Java test source files, of which full
+     * qualified names end with "Test" and the full names of corresponding modules are
+     * in the extract module info map.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a test of checkstyle module
+     */
+    public static boolean isCheckstyleModuleTest(GitChange change) {
+        final boolean returnValue;
+        if (isJavaTestSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change);
+            if (fullName.endsWith(TEST_POSTFIX)) {
+                returnValue = NAME_TO_MODULE_EXTRACT_INFO.containsKey(
+                        fullName.substring(0, fullName.length() - TEST_POSTFIX.length()));
+            }
+            else {
+                returnValue = false;
+            }
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a Java test source file.
+     * Changes of Java test source files are which have path matching
+     * {@code JAVA_SOURCE_PARTTEN} and in "test" directory.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a Java test source file.
+     */
+    private static boolean isJavaTestSource(GitChange change) {
```

Comment:
One last thing, please group `isJava...` methods together. Do the same for `isCheckstyle...` methods.